components/worktemplates/components/preview.tsx

```diff
@@ -59,7 +106,14 @@ const Preview = ({template, ...props}: PreviewProps) => {
                     key={'channels'}
                     message={template.description.channel.message}
                     items={channels}
-                    onUpdateIllustration={setCurrentIllustration}
+                    onUpdateIllustration={(id: string, illustration: string) => setIllustrationDetails({
+                        prior: {...illustrationDetails.current},
+                        current: {
+                            mounted: false,
+                            type: TemplateType.CHANNELS,
+                            illustration,
+                        },
+                    })}
```

Comment:
Also we have this bit of code in a lot of places, it might be worth it to make a method to improve readability.

```suggestion
                    onUpdateIllustration={handleUpdateIllustration}
```